src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/CustomUserDetailsService.java

```diff
@@ -1,41 +1,27 @@
 package uk.gov.hmcts.reform.sandl.snlapi.security;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sandl.snlapi.repositories.UserRepository;
 import uk.gov.hmcts.reform.sandl.snlapi.security.model.User;
 import uk.gov.hmcts.reform.sandl.snlapi.security.model.UserPrincipal;
 
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
-
 @Service
 public class CustomUserDetailsService implements UserDetailsService {
 
-    private final Map<String, User> users;
-
-    CustomUserDetailsService() {
-        Map<String, User> usersToAdd = new HashMap<>();
-        usersToAdd.put("officer1", new User("officer1", "asd", "Officer 1",
-            Arrays.asList("USER", "OFFICER")));
-        usersToAdd.put("judge1", new User("judge1", "asd", "John Harris",
-            Arrays.asList("USER", "JUDGE")));
-        usersToAdd.put("admin", new User("admin", "asd", "Administrator man",
-            Arrays.asList("USER", "ADMIN")));
-
-        this.users = Collections.unmodifiableMap(usersToAdd);
-    }
+    @Autowired
+    private UserRepository userRepository;
 
     @Override
-    public UserDetails loadUserByUsername(String usernameOrEmail) {
+    public UserDetails loadUserByUsername(String username) {
         try {
-            User user = users.get(usernameOrEmail);
+            User user = userRepository.findByUsername(username);
             return UserPrincipal.create(user);
         } catch (NullPointerException ex) {
-            throw new UsernameNotFoundException("User not found with username or email : " + usernameOrEmail, ex);
+            throw new UsernameNotFoundException("User not found with username: " + username, ex);
```

Comment:
may this message leak to frontend? see https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Authentication_and_Error_Messages

Reply:
Good point although I didn't introduce this message (just removed the email bit), I'll change to make the message more generic

Reply:
Hmmm... if this can possibly leak to the front end, I guess it's possible for the exception name to also leak, which then is a problem in itself

Reply:
This is by design in Spring, and some other Spring implementations of UserDetails include the username that wasn't found in the message. So I'm assuming this doesn't leak through to the frontend, as it would have undoubtedly been flagged as a Spring issue.